api/clients/eigenda_client.go

```diff
@@ -16,13 +16,19 @@ import (
 	"github.com/ethereum/go-ethereum/log"
 )
 
+// IEigenDAClient is a wrapper around the DisperserClient interface which
+// encodes blobs before dispersing them, and decodes them after retrieving them.
 type IEigenDAClient interface {
 	GetBlob(ctx context.Context, batchHeaderHash []byte, blobIndex uint32) ([]byte, error)
 	PutBlob(ctx context.Context, txData []byte) (*grpcdisperser.BlobInfo, error)
 	GetCodec() codecs.BlobCodec
 }
 
+// EigenDAClient is a wrapper around the DisperserClient which
+// encodes blobs before dispersing them, and decodes them after retrieving them.
 type EigenDAClient struct {
+	// TODO: all of these should be private, to prevent users from using them directly,
+	// which breaks encapsulation and makes it hard for us to do refactors or changes
```

Comment:
this I personally think is a refactor worth doing... but it might break some client code... :\

Reply:
agree this should be private. From my experience, it is difficult to understand data semantics in each stage or encoding and decoding.

Reply:
Don't understand your comment, why are you mentioning data semantics here?
And are you saying I should change the fields to private right now? Are we not scared of breaking user code that might be using these?

Reply:
just from the developers' view, it tool me a while to understand the intermediate phases. If someone else would use it, I doubted it would be correct

Reply:
Note: we decided to do all the breaking changes in a future PR.